File: CheckersOnline.Server/CheckersOnline.Server/Models/GameState.cs
namespace CheckersOnline.Server.Models
{
    public class GameState
    {
        public Piece?[][] Board { get; set; } = CreateEmptyBoard();
        public PieceColor CurrentTurn { get; set; }

        private static Piece?[][] CreateEmptyBoard()
        {
            var board = new Piece?[8][];
            for (int i = 0; i < 8; i++)
                board[i] = new Piece?[8];
            return board;
        }
    }
}

File: CheckersOnline/src/models/GameState.ts
export interface GameState {
  board: (Piece | null)[][];
  currentTurn: PieceColor;
}

export interface Piece {
  color: PieceColor;
  type: PieceType;
}

export type PieceColor = "Black" | "White";
export type PieceType = "Man" | "King";
